python/lang/security/audit/dangerous-asyncio-exec-tainted-env-args.yaml

```diff
@@ -0,0 +1,86 @@
+                          - pattern: <... $ARGS ...>
+                      - patterns:
+                          - pattern-inside: |
+                              $PARSER = optparse.OptionParser(...)
+                              ...
+                          - pattern-inside: |
+                              $ARGS = $PARSER.parse_args()
+                          - pattern: <... $ARGS ...>
+                      - patterns:
+                          - pattern-either:
+                              - pattern-inside: |
+                                  $OPTS, $ARGS = getopt.getopt(...)
+                                  ...
+                              - pattern-inside: |
+                                  $OPTS, $ARGS = getopt.gnu_getopt(...)
+                                  ...
+                          - pattern-either:
+                              - patterns:
+                                  - pattern-inside: |
+                                      for $O, $A in $OPTS:
+                                        ...
+                                  - pattern: $A
+                              - pattern: $ARGS
+    pattern-sinks:
+      - pattern-either:
+          - patterns:
+              - pattern-not: $LOOP.subprocess_exec($PROTOCOL, "...", ...)
+              - pattern-not: $LOOP.subprocess_exec($PROTOCOL, ["...",...], ...)
+              - pattern: $LOOP.subprocess_exec(...)
+          - patterns:
+              - pattern-not: $LOOP.subprocess_exec($PROTOCOL, "=~/(sh|bash|ksh|csh|tcsh|zsh)/", "-c", "...", ...)
+              - pattern: $LOOP.subprocess_exec($PROTOCOL, "=~/(sh|bash|ksh|csh|tcsh|zsh)/", "-c",...)
+          - patterns:
+              - pattern-not: $LOOP.subprocess_exec($PROTOCOL, ["=~/(sh|bash|ksh|csh|tcsh|zsh)/", "-c", "...", ...], ...)
+              - pattern: $LOOP.subprocess_exec($PROTOCOL, ["=~/(sh|bash|ksh|csh|tcsh|zsh)/", "-c", ...], ...)
+    message: >-
+      Detected subprocess function '$LOOP.subprocess_exec' with user controlled data.
+      You may consider using 'shlex.escape()'.
+    metadata:
+      owasp:
+        - A03:2020 - Injection
+        - A01:2017 - Injection
+      cwe: "CWE-78: Improper Neutralization of Special Elements used in an OS Command ('OS Command Injection')"
+      asvs:
+        section: "V5: Validation, Sanitization and Encoding Verification Requirements"
+        control_id: 5.3.8 OS Command Injection
+        control_url: https://github.com/OWASP/ASVS/blob/master/4.0/en/0x13-V5-Validation-Sanitization-Encoding.md#v53-output-encoding-and-injection-prevention-requirements
+        version: "4"
+      references:
+        - https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.subprocess_exec
+        - https://docs.python.org/3/library/shlex.html
+        - https://semgrep.dev/docs/cheat-sheets/python-command-injection/
+      category: security
+      technology:
+        - python
+      confidence: HIGH
```

Comment:
I think people are going to argue about confidence of environment variables and OS commands as high confidence, because its usually 'defined by the consumer of the applicaation

Reply:
Yeah, I wanted to convey the fact that confidence is high related to the taint flow being real/accurate (vs a simpler non-taint based rule) but I agree it can be misleading. I lowered the confidence for all audit rules in b775beb85fa56f2a688d0d36cdf81ca090bc3d9f